src/macaron/artifact/local_artifact.py

```diff
@@ -0,0 +1,275 @@
+# Copyright (c) 2024 - 2024, Oracle and/or its affiliates. All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.
+
+"""This module declares types and utilities for handling local artifacts."""
+
+import fnmatch
+import glob
+import os
+from collections.abc import Mapping
+
+from packageurl import PackageURL
+
+from macaron.artifact.maven import construct_maven_repository_path
+
+
+def construct_local_artifact_paths_glob_pattern_maven_purl(maven_purl: PackageURL) -> list[str] | None:
+    """Return a list of glob pattern(s) representing maven artifacts in a local maven repository.
+
+    The glob pattern(s) can be used to search in `<...>/.m2/repository` directory.
+
+    Parameters
+    ----------
+    maven_purl : PackageURL
+        A maven type PackageURL instance.
+
+    Returns
+    -------
+    list[str] | None
+        A list of glob patterns or None if an error happened.
+
+    Examples
+    --------
+    >>> from packageurl import PackageURL
+    >>> purl = PackageURL.from_string("pkg:maven/com.oracle.macaron/macaron@0.13.0")
+    >>> construct_local_artifact_paths_glob_pattern_maven_purl(purl)
+    ['com/oracle/macaron/macaron/0.13.0']
+    """
+    if not maven_purl.type == "maven":
+        return None
+
+    group = maven_purl.namespace
+    artifact = maven_purl.name
+    version = maven_purl.version
+
+    if group is None or version is None:
+        return None
+
+    return [construct_maven_repository_path(group, artifact, version)]
+
+
+def construct_local_artifact_paths_glob_pattern_pypi_purl(pypi_purl: PackageURL) -> list[str] | None:
+    """Return a list of glob pattern(s) representing python artifacts in a virtual environment.
+
+    The glob pattern(s) can be used to search in `<...>/<python_venv>/lib/python3.x/site-packages`
+    directory.
+
+    Parameters
+    ----------
+    maven_purl : PackageURL
```

Comment:
```suggestion
    pypi_purl : PackageURL
```

Reply:
Fixed in https://github.com/oracle/macaron/pull/864/commits/da0f514376752d38af26e2be396c6e31001a7a24